### PR TITLE
Add failing test to show S3 md5 null string issue

### PR DESCRIPTION
**Pull request by bskarda**
Issue is outlined in
https://github.com/localstack/localstack/issues/869
and
https://github.com/localstack/localstack/issues/538
(Specifically
https://github.com/localstack/localstack/issues/538#issuecomment-424379644)

One of the external symptoms of the bug is that S3 interactions for
objects that are only the null string (zero length objects) throw
an error like
`Client calculated content hash (contentMD5: 1B2M2Y8AsgTpgAmY7PhCfg== in base 64) didn't match hash (etag: ...`
which is telling because the `contentMD5` value is the base64 encoded
value for the md5 of null string. The correct etag value should be
`d41d8cd98f00b204e9800998ecf8427e`.

On multiple executions of the test case the etag generated by the S3
server is different so there is some non deterministic value being put
into the hash function. I haven't been able to find where the actual
issue exists in the code as I don't use python very often.

One workaround for avoiding this issue is to disable the MD5 validations
in the S3 client. For Java that means setting the following properties
to any value.
```
com.amazonaws.services.s3.disableGetObjectMD5Validation
com.amazonaws.services.s3.disablePutObjectMD5Validation
```

**Please refer to the contribution guidelines in the README when submitting PRs.**

### CODE REVIEW

File: localstack/ext/java/src/test/java/cloud/localstack/S3UploadTest.java [hunk 1]
```diff
@@ -49,6 +50,30 @@ public void testGreaterThan128k() throws Exception  {
...
 		testUpload(String.join("", Collections.nCopies(13107, "abcdefghij"))); // Just slightly less than 2^17 bytes
+    }
+
+	/**
+	 * Tests upload of empty file. This is an operation that hadoop's S3AFilesystem executes to create "directories"
+	 * in S3.
+	 *
+	 * This test is currently failing because the S3 server doesn't properly calculate the MD5 of null string.
+	 */
+	@Test
```
Comment (bskarda) [inline localstack/ext/java/src/test/java/cloud/localstack/S3UploadTest.java]: I'm happy to throw an `@Ignored` on here to make the CI pass. I mainly wanted to explicitly bring this issue up in a PR since it seems like there are a number of open issues around this bug.

**Comment by whummer** [comment]
Thanks for reporting @bskarda - this appears to be fixed with the changes in #1140

**Comment by Pedrock** [comment]
@whummer Shouldn't the test be added in order to prevent a regression?

**Comment by whummer** [comment]
@Pedrock You're absolutely right! Not sure what I was thinking when closing the PR. :) Thanks for pointing that out.
